db/migrate/20241008053852_add_external_voucher_id_external_voucher_set_id_voucher_type_to_vouchers.rb

```diff
@@ -0,0 +1,7 @@
+class AddExternalVoucherIdExternalVoucherSetIdVoucherTypeToVouchers < ActiveRecord::Migration[7.0]
+  def change
+    add_column :vouchers, :external_voucher_id, :uuid
+    add_column :vouchers, :external_voucher_set_id, :uuid
+    add_column :vouchers, :voucher_type, :string
```

Comment:
How about we create a new `Vouchers::Vine` model and use the existing `type` field?

Reply:
A vine voucher really is a specific type of `FlatRate` voucher, so it didn't make sense to use the `type` field. In the original  specifications for voucher, it talked about having a voucher type to differentiate between promotion, food equity etc... . I fee like a vine voucher fits this model better than a voucher that is calculated differently.

Reply:
It still feels strange to have some very specific behaviour associated with the `voucher_type`. I worry that we may lose site of it when developing other areas of the app, and forget to differentiate between `vine` and `local`.

That's why I would feel more comfortable with a sub-type, maybe even `Vouchers::FlatRate::Vine`? 
But.. that doesn't really help, we will still need to differentiate between `vine` and `local`. 

Hmm, I'm probably not seeing the bigger picture. Time to review more of the PR!

Reply:
Yeah it's not the best, we are trying to shoehorn VINE voucher into a model that wasn't really design for that, on top of voucher already been shoehorned into adjustments :sweat_smile: